src/telemetry/telemetryLogger.ts

```diff
@@ -65,25 +59,28 @@ export function getTelemetryLogger(): vscode.TelemetryLogger {
     }
     analytics = new Analytics({ writeKey, disable: false });
   }
-  // We extract the vscode session ID from the event data, but this random id will be sent if it is undefined (unlikely but not guranteed by the type def)
+
   segmentAnonId = randomUUID();
+
   telemetryLogger = vscode.env.createTelemetryLogger({
     sendEventData: (eventName, data) => {
-      const cleanEventName = eventName.replace(/^confluentinc\.vscode-confluent\//, ""); // Remove the prefix that vscode adds to event names
+      // Remove the prefix that vscode adds to event names
+      const cleanEventName = eventName.replace(/^confluentinc\.vscode-confluent\//, "");
+      // Extract & save the user id if was sent
       if (data?.user?.id) userId = data.user.id;
-      if (data?.identify && userId) {
+      if (data?.identify && data?.user) {
         analytics?.identify({
           userId,
-          anonymousId: data?.["common.vscodesessionid"] || segmentAnonId,
-          traits: { email: data?.user?.username, social_connection: data?.user?.social_connection },
+          anonymousId: segmentAnonId,
+          traits: { ...data.user },
         });
         // We don't want to send the user traits or identify prop in the following Track call
         delete data.identify;
         delete data.user;
       }
       analytics?.track({
         userId,
-        anonymousId: data?.["common.vscodesessionid"] || segmentAnonId,
```

Comment:
Why this change?

Reply:
Just setting us up to handle logout in near future... We're already sending `vscodesessionid` with the common properties in data, and the `segmentAnonId` is stable in each instance of the TelemetryLogger, so it won't make much difference now but in updates I'll be able to reset the anon id (during log out flow or other cases where the User Id might be new for an existing VSCode session)